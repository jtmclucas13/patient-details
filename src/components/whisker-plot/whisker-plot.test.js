import React from 'react';
import { shallow } from 'enzyme';
import WhiskerPlot from './whisker-plot';

describe('component WhiskerPlot', () => {
    test('renders as expected', () => {
        const wrapper = renderWhiskerPlot();

        expect(wrapper).toMatchSnapshot();
    });
});

/**
* Helper function for rendering WhiskerPlot in tests without boilerplate required props
* @param {Object} [props={}] props for WhiskerPlot
* @return {ShallowWrapper} enzyme wrapper for WhiskerPlot 
*/
function renderWhiskerPlot(props = {}) {
    const propsToUse = {
        allValues: [1, 10],
        targetValue: 5,
        ...props,
    };

    return shallow(<WhiskerPlot {...propsToUse} />);
}
