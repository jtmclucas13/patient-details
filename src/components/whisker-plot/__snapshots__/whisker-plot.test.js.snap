// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component WhiskerPlot renders as expected 1`] = `
<svg
  className="whisker-plot"
  height={20}
  width={200}
>
  <line
    stroke="black"
    strokeWidth={2}
    x1={0}
    x2={200}
    y1={10}
    y2={10}
  />
  <line
    stroke="black"
    strokeWidth={2}
    x1={1}
    x2={1}
    y1={0}
    y2={20}
  />
  <line
    stroke="black"
    strokeWidth={2}
    x1={199}
    x2={199}
    y1={0}
    y2={20}
  />
  <line
    stroke="black"
    strokeWidth={2}
    x1={100}
    x2={100}
    y1={0}
    y2={20}
  />
  <circle
    cx={88.88888888888889}
    cy={10}
    r={10}
  />
</svg>
`;
